### PR TITLE
feat(FOROME-732): add all, not logical operators for enum panel

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-732

PR into feat/FOROME-365

### CODE REVIEW

File: src/store/filter.ts [hunk 1]
```diff
@@ -20,6 +21,8 @@ interface AddSelectedFiltersI {
   group: string
   groupItemName: string
   variant?: TVariant
+  variants?: any[]
```
Comment (AndreyUstyumenko) [inline src/store/filter.ts]: TVariant[]?

**Review by AlMaQntr — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: src/pages/filter/ui/all-not-mode-view.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+export const AllNotModeView = observer(
+  ({ filters }: AllNotModeViewProps): ReactElement => {
+    const filtersWithAllMode = useMemo(() => {
+      return Object.keys(filters).includes('All')
+    }, [filters])
+
+    const filtersWithNotMode = useMemo(() => {
+      return Object.keys(filters).includes('Not')
+    }, [filters])
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/all-not-mode-view.tsx]: Convert this to a dumb component that receives only 'not' or 'all' parameter. Or boolean.

File: src/pages/filter/ui/filterAttributes.store.ts [hunk 3]
```diff
@@ -49,6 +58,16 @@ export class FilterAttributesStore {
     return toJS(this.getDatasetEnumValues(this.currentGroup))
   }
 
+  toggleMode(modeType: string): void {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/filterAttributes.store.ts]: modeType: ModeTypes

File: src/pages/filter/ui/filterAttributes.store.ts [hunk 4]
```diff
@@ -49,6 +58,16 @@ export class FilterAttributesStore {
...
+    this.isAllMode = modeType === ModeTypes.All
+    this.isNotMode = modeType === ModeTypes.Not
+  }
+
+  resetModes(): void {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/filterAttributes.store.ts]: add access modifiers everywhere

File: src/pages/filter/ui/filterAttributes.store.ts [hunk 5]
```diff
@@ -25,6 +27,9 @@ export class FilterAttributesStore {
   private datasetStore: DatasetStore
   private filterStore: FilterStore
 
+  isAllMode = false
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/filterAttributes.store.ts]: better to create a property 'currentMode: ModeTypes' therefore you won't need to keep a weird boolean prop.
I could be wrong, pls check it

File: src/pages/filter/ui/filterAttributes.store.ts [hunk 6]
```diff
@@ -104,6 +128,15 @@ export class FilterAttributesStore {
         selectedSubAttributes[variant[0]] = variant[1]
       }
     })
+
+    if (this.isAllMode) {
+      selectedSubAttributes['All'] = 1
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/filterAttributes.store.ts]: If you had 'currentMode' all you had to do is `selectedSubAttributes[this.currentMode] = 1`

File: src/utils/getModeType.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+export const getModeType = (
+  isAllMode: boolean,
+  isNotMode: boolean,
+): ConditionJoinMode => {
+  return isAllMode
+    ? ConditionJoinMode.AND
+    : isNotMode
+    ? ConditionJoinMode.NOT
+    : ConditionJoinMode.OR
```
Comment (AndreyUstyumenko) [inline src/utils/getModeType.ts]: do an if else condition, hard to read

File: src/store/filter.ts [hunk 8]
```diff
@@ -66,6 +73,17 @@ export class FilterStore {
     if (variant) {
       this.selectedFilters[group][groupItemName][variant[0]] = variant[1]
     }
+
+    // temporarilly, remove this agr begin required
+    if (modeTypes) {
```
Comment (AndreyUstyumenko) [inline src/store/filter.ts:78]: const [isAllMode, isNotMode] = modeTypes

File: src/store/filter.ts [hunk 9]
```diff
@@ -66,6 +73,17 @@ export class FilterStore {
...
+      if (modeTypes[0]) {
```
Comment (AndreyUstyumenko) [inline src/store/filter.ts]: `const [isAllMode, isNotMode] = modeTypes`
avoid array[index] constructions they are confusing

File: src/store/filter.ts [hunk 10]
```diff
@@ -19,7 +20,8 @@ export type SelectedFiltersType = Record<
 interface AddSelectedFiltersI {
   group: string
   groupItemName: string
-  variant?: TVariant
+  variant: TVariant
+  modeTypes?: boolean[]
```
Comment (AndreyUstyumenko) [inline src/store/filter.ts]: modeTypes?: [boolean, boolean]

File: src/pages/filter/ui/selected-filter-card.tsx [hunk 11]
```diff
@@ -16,7 +17,12 @@ interface Props {
 export const SelectedFilterCard = observer(
   ({ title, filters, onRemove }: Props): ReactElement => {
     const [isOpen, open, close] = useToggle(true)
-    const filtersEntries = Object.entries(filters)
+
+    const filtersEntries = useMemo(() => {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/selected-filter-card.tsx]: the purpose of this function is not obvious. Rename to something more clear

File: src/pages/filter/ui/query-builder/ui/all-not-mods.tsx [hunk 12]
```diff
@@ -21,20 +23,26 @@ export const AllNotMods = observer(
...
     <div className="flex text-14 text-blue-bright">
-      <div className="flex items-center">
-        <Checkbox
-          checked={isAllModeChecked ?? false}
-          className="mr-1 cursor-pointer"
-          disabled={isAllModeDisabled}
-          onChange={toggleAllMode}
-        />
+      {(groupSubKind === SubKindsEnum.Multi ||
+        groupSubKind === SubKindsEnum.InheritanceZ) && (
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/ui/all-not-mods.tsx]: create a boolean is* variable for this binary expression

File: src/pages/filter/ui/panels/function-panel/function-panel.store.ts [hunk 13]
```diff
@@ -70,20 +70,25 @@ class FunctionPanelStore {
     await datasetStore.setConditionsAsync([conditions], 'func')
   }
 
-  public addSelectedFilters(variant: TVariant): void {
+  public addSelectedFilters(variant: TVariant, modeTypes?: boolean[]): void {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/panels/function-panel/function-panel.store.ts]: [boolean, boolean]

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**